packages/analytics-js/src/components/configManager/util/validate.ts

```diff
@@ -37,10 +39,25 @@ const isValidSourceConfig = (res: any): boolean =>
 const isValidStorageType = (storageType?: StorageType): boolean =>
   typeof storageType === 'string' && SUPPORTED_STORAGE_TYPES.includes(storageType);
 
+const validateAndReturnCookieServerUrl = (
+  useServerSideCookies?: boolean,
+  dataServerUrl?: string,
+  logger?: ILogger,
+) => {
+  if (useServerSideCookies && dataServerUrl) {
+    if (isValidUrl(dataServerUrl)) {
+      return dataServerUrl;
+    }
+    logger?.error(COOKIE_SERVER_URL_INVALID_ERROR('dataServerUrl'));
+  }
+  return 'invalid';
```

Comment:
How is this return value helpful?

Reply:
The data server URL is not mandatory. We have to infer it from the data plane URL if it's not provided.

Reply:
Should we disable the server side cookies feature if the provided server URL is invalid?

Reply:
Addressed